examples/scripts/ExampleMultipleChaperones.py

```diff
@@ -1,64 +1,69 @@
 
-numOpen = 1
-numClosed = 1
+numOp = 1
+numCl = 1
+num = 2
 pwd = os.path.dirname(__file__)
 
-# Create a particle object
+# create particle object(s)
 particleOp = ps.Particle()
 particleOp.read_pdb(os.path.join(pwd,'../input/pdb/3iyf.pdb'), ff='WK')
 
-
 particleCl = ps.Particle()
 particleCl.read_pdb(os.path.join(pwd,'../input/pdb/3j03.pdb'), ff='WK')
 
-# Load beam
+# load beam
 beam = ps.Beam(os.path.join(pwd,'../input/beam/amo86615.beam'))
 
 geom = os.path.join(pwd,'../input/lcls/amo86615/PNCCD::CalibV1/Camp.0:pnCCD.1/geometry/0-end.data')
 
-# Load and initialize the detector
-det = ps.PnccdDetector(geom=geom, beam = beam)
+# load and initialize the detector
+det = ps.PnccdDetector(geom=geom, beam=beam)
 
 tic = time.time()
-patternOp = det.get_photons(device='gpu', particle=particleOp)
+experiment = ps.FXSExperiment(det, beam, [particleOp], numOp)
+patternOp = experiment.generate_image_stack()
 toc = time.time()
 print("It took {:.2f} seconds to finish SPI calculation.".format(toc-tic))
 
-patternCl = det.get_photons(device='gpu', particle=particleCl)
+experiment = ps.FXSExperiment(det, beam, [particleCl], numCl)
+patternCl = experiment.generate_image_stack()
 
-# Calculates 1 diffraction pattern from 1 open chaperones + 1 closed chaperones
-pattern = det.get_fxs_photons_slices(device='gpu', beam_focus_radius=beam.focus_xFWHM/2, jet_radius=1e-4, mesh_length=151, particles={particleOp:numOpen,particleCl:numClosed})
+# calculate 1 diffraction pattern from 2 particles, where each particle has 50% chance of being Open or Closed
+# (end up in 25% with two Open, 25% with two Closed, and 50% with one of each)
+experiment = ps.FXSExperiment(det, beam, [particleOp, particleCl], num)
+pattern = experiment.generate_image_stack()
 
 pattern_no_polarization = det.remove_polarization(pattern, res=None)
 np_img = det.assemble_image_stack(pattern_no_polarization)
 mask = np.ones_like(pattern_no_polarization)
 np_mask = det.assemble_image_stack(mask)
 
 # write data to HDF5
-#with h5.File('mixed_chaperones_and_mask.hdf5', 'w') as f:
-with h5.File('test.hdf5', 'w') as f:
+with h5.File('mixed_chaperones_and_mask.hdf5', 'w') as f:
     f.create_dataset('img', data=np_img)
     f.create_dataset('mask', data=np_mask, dtype=np.int16)
 
 fig = plt.figure(figsize=(10, 8))
 plt.subplot(131)
-plt.imshow(det.assemble_image_stack(patternOp),vmin=0, vmax=10)
-plt.title('Open state')
+plt.imshow(det.assemble_image_stack(patternOp),norm=LogNorm())
+plt.colorbar()
+plt.title('Open')
 plt.subplot(132)
-plt.imshow(det.assemble_image_stack(patternCl),vmin=0, vmax=10)
-plt.title('Closed state')
+plt.imshow(det.assemble_image_stack(patternCl),norm=LogNorm())
+plt.colorbar()
+plt.title('Closed')
 plt.subplot(133)
-plt.imshow(det.assemble_image_stack(pattern),vmin=0, vmax=10)
-plt.title('%i Open + %i Closed'%(numOpen,numClosed))
+plt.imshow(det.assemble_image_stack(pattern),norm=LogNorm())
+plt.colorbar()
+plt.title('Mixed')
```

Comment:
In title state how many open and how many closed

Reply:
in a new subplot, imshow np_mask with title "polarization corrected region"

Reply:
ok, that sounds good for now.
Later, we should introduce a way to assign a probability for each particle.

Reply:
We can assign a probability for each particle by passing in values to ratios. (default=None)
`experiment = ps.FXSExperiment(det, beam, particles, n_part_per_shot, ratios=None)`